### DISCUSSION TITLE
Running an executable file with parameters from TI Process

### DISCUSSION THREAD

**Discussion by ishmeets920** [cubewise-code/tm1py#759] [Q&A] [answered]
Hello @MariusWirtz, 
I need to run an exe file i created in python to transfer a data from one cube to another. 

I am using argparse library of python script to have parameters at run time. 
>`Parameter = '"-n  parameter"';

For to process, 
I have used `executecommand(Expand(exepath|parameter),1)`;

Issue1:
My ti process is just not taking the parameters i enter. Please help

**Python Code**:
```
import sys
 import logging
 import argparse
 path = '\\tm1server\ImM\Transfer\logs.txt'
 logging.basicConfig(filename=path,
                     filemode='a',
                     format='%(asctime)s,%(msecs)d %(name)s %(levelname)s %(message)s',
                     datefmt='%H:%M:%S',
                     level=logging.DEBUG)
 
 logging.info("Came inside the code")
 logger = logging.getLogger('Log 1')
 
 File = sys.argv[1]
 logging.info("Here is the file - " + File)
 logger = logging.getLogger('Log 2')

Converting the python to exe :
Pyinstaller --hiddenimport win32timezone -F Check.py

```

**Here is the TI code:**
```
cmd = '"\\Data_Transfer\Check.exe"';
file= '"\Data_Transfer\config.ini"';
hello = '"hello"';

sRun = 'cmd %file% %hello%';

ExecuteCommand(Expand(sRun),1);
```

**Actual result:**
It prints 'came inside the code' 

**Expected Result:**
Should print 
Came inside the code
Data_Transfer\config.ini
hello

**Comment by MariusWirtz** [comment]
Does the python script itself work fine when executed from command instead of from TI?

Does the pythons script still work when compiled to an exe?

Next thing you can look into is if the actual command is fine. Do an `ItemReject(Expand(sRun))`. Then copy the command into the command prompt and see if it does the right thing. If not troubleshoot it in the command prompt first.

**Reply by ishmeets920** [reply]
Hi @MariusWirtz,

Thanks for replying.

Yes the exe file itself works. 

When tried doing ItemReject it shows me the server path of exe file and server path of the config file i am passing as the parameter. 

In commandprompt at local machine,
I am able to run the exe with parameter with no issue.

**Reply by ishmeets920** [reply]
Do we have any sample code for running exe file with parameter (config.ini file) ?

**Reply by ishmeets920** [reply]
Hi @MariusWirtz,
Thanks it worked the path I was trying to run was not correct. Worked now.

**Comment by rkvinoth** [comment]
Pass the full path including the drive in which the executable is residing.

```
cmd = 'D:\Data_Transfer\Check.exe';
file= 'D:\Data_Transfer\config.ini';
hello = 'hello';

sRun = '"cmd" "%file%" "%hello%"';

ExecuteCommand(Expand(sRun), 1);
```

**Reply by ishmeets920** [reply]
Hi @rkvinoth,
Thanks for replying.

I tried running this way but it says the exe file path is not specified.

**Comment by rkvinoth** [comment, marked as answer]
```
cmd = 'D:\Data_Transfer\Check.exe';
file= 'D:\Data_Transfer\config.ini';
hello = 'hello';

sRun = '"cmd" "%file%" "%hello%"';

LOGOUTPUT('INFO', sRun);

ExecuteCommand(Expand(sRun), 1);
```

copy the command from tm1server.log and try running it in command prompt.
If that works, the problem could be because of the account which is running the TM1 instance.

**Reply by ishmeets920** [reply]
Thanks @rkvinoth, it worked
